### ISSUE TITLE
W0223 does not get raised correctly

### ISSUE THREAD

**Issue by alibeyram** [PyCQA/pylint#9979]
### Bug description

The not implemented warning does not get raised 

here is a simple code that shows it 

```python
from abc import ABC, abstractmethod


class Base(ABC):
    @abstractmethod
    def do_something(self):
        pass


class Derived(Base):
    def do_something_else(self):
        pass

```

One way to test it is to recreate `abc.ABC` and use that for inheriting

 ```python
from abc import ABCMeta, abstractmethod


class MYABC(metaclass=ABCMeta):
    __slots__ = ()


class Base(MYABC):
    @abstractmethod
    def do_something(self):
        pass


class Derived(Base):  #### the W0223:abstract-method raised here
    def do_something_else(self):
        pass
```
```


### Configuration

_No response_

### Command used

```shell
pylint test.py
```

### Pylint output

```python
Your code has been rated at 10.00/10
```

### Expected behavior

W0223: Method 'do_something' is abstract in class 'Base' but is not overridden in child class 'Derived' (abstract-method)

### Pylint version

```shell
pylint 3.3.1
astroid 3.3.4
Python 3.12.5 | packaged by conda-forge | (main, Aug  8 2024, 18:32:50) [Clang 16.0.6 ]
```

**Comment by alibeyram** [comment]
@mbyrnepr2 are you gonna merge the PR?

**Comment by mbyrnepr2** [comment]
After looking into this further, and with the helpful output of the home-assistant primer in the CI, I think the original example is not a false negative since we don't know if the `Derived` subclass will be instantiated or not. If we emit `abstract-method` for `Derived` we get potentially a lot of false positives as can be seen from the home-assistant output.

Here is a manufactured example which is something like the home-assistant structure where we have subclasses of the original abc.ABC class, each implementing some of abstractmethods:

```python
# Python is happy to instantiate Class3 since it implements all 3 abstractmethods.
import abc


class Base(abc.ABC):
    @abc.abstractmethod
    def method1(self):
        print("abstract")

    @abc.abstractmethod
    def method2(self):
        print("abstract")

    @abc.abstractmethod
    def method3(self):
        print("abstract")


class Class1(Base):
    def method1(self):
        print("method1 concrete")


class Class2(Class1):
    def method2(self):
        print("method2 concrete")


class Class3(Class2):
    def method3(self):
        print("method3 concrete")


myclass = Class3()
```

I'll change the label to `false positive` for the second example that was provided:
```python
from abc import ABCMeta, abstractmethod


class MYABC(metaclass=ABCMeta):
   __slots__ = ()


class Base(MYABC):
   @abstractmethod
   def do_something(self):
       pass


class Derived(Base):  #### the W0223:abstract-method raised here
   def do_something_else(self):
       pass
```

**Comment by mbyrnepr2** [comment]
Also looks like a duplicate of https://github.com/pylint-dev/pylint/issues/7950
